src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ModuleInitializerAttributeShouldNotBeUsedInLibrariesTests.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ModuleInitializerAttributeShouldNotBeUsedInLibraries,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class ModuleInitializerAttributeShouldNotBeUsedInLibrariesTests
+    {
+        [Theory]
+        [InlineData("public", "public", false)]
+        [InlineData("public", "internal", false)]
+        [InlineData("internal", "public", false)]
+        [InlineData("internal", "internal", false)]
+        [InlineData("public", "public", true)]
+        [InlineData("public", "internal", true)]
+        [InlineData("internal", "public", true)]
+        [InlineData("internal", "internal", true)]
+        public async Task CA2255ModuleInitializerOnMethod(string classModifier, string methodModifier, bool useAsync)
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        @$"
+                        {classModifier} class AccessibleClass
+                        {{
+                            [[|System.Runtime.CompilerServices.ModuleInitializer|]]
+                            {methodModifier} static {(useAsync ? "async" : "")} void AccessibleInitializer() {{ }}
+                        }}
+                        "
+                    }
+                },
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9
+            }.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("public", "public", false)]
+        [InlineData("public", "internal", false)]
+        [InlineData("internal", "public", false)]
+        [InlineData("internal", "internal", false)]
+        [InlineData("public", "public", true)]
+        [InlineData("public", "internal", true)]
+        [InlineData("internal", "public", true)]
+        [InlineData("internal", "internal", true)]
+        public async Task CA2255ModuleInitializerOnMethod_WithParens(string classModifier, string methodModifier, bool useAsync)
```

Comment:
NIT: I might miss something but don't see any difference between this and the previous test, what is `Parens`?

Reply:
It's subtle:

```csharp
[[|System.Runtime.CompilerServices.ModuleInitializer|]]
// vs
[[|System.Runtime.CompilerServices.ModuleInitializer()|]]
```

I mostly added this test because I wasn't sure (and was curious) if the diagnostic would highlight just the attribute type name or the entire attribute; it highlights the entire attribute, parens/arguments included.